pilot/pkg/model/mesh.go

```diff
@@ -0,0 +1,785 @@
+// Copyright 2018 Istio Authors
```

Comment:
Can you pull all of the Endpoints type defs and functions into a new `endpoints.go` file? Right now the methods of `Mesh` and `Endpoints` are mixed up and it makes it harder to read and understand what either does (especially since Mesh functions need to be mutex protected but Endpoint methods don't).